src/dijkstra.rs

```diff
@@ -0,0 +1,105 @@
+use std::cmp::Ordering;
+use std::collections::BinaryHeap;
+
+#[derive(Clone)]
+pub struct Edge {
+    pub to: usize,
+    pub cost: usize,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq)]
+struct State {
+    cost: usize,
+    position: usize,
+}
+
+// The priority queue depends on `Ord`.
+// Explicitly implement the trait so the queue becomes a min-heap
+// instead of a max-heap.
+impl Ord for State {
+    fn cmp(&self, other: &State) -> Ordering {
+        // Notice that the we flip the ordering on costs.
+        // In case of a tie we compare positions - this step is necessary
+        // to make implementations of `PartialEq` and `Ord` consistent.
+        other
+            .cost
+            .cmp(&self.cost)
+            .then_with(|| self.position.cmp(&other.position))
+    }
+}
+
+// `PartialOrd` needs to be implemented as well.
```

Comment:
`Ord` requires that the type also be `PartialOrd` and `Eq` (which requires `PartialEq`).
For more details, refer to the document:
https://doc.rust-lang.org/std/cmp/trait.Ord.html#how-can-i-implement-ord